File: tests/resources/bid/1/prog/bid.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

constexpr int SIZE = 1e6 + 10;
using Vertex = int;

string bida(vector<Vertex> graph[], int n){
	for (int i = 0; i < n; i++){
		if (graph[i].size() % 2 != 0)
			return "NIE";
	}
	return "TAK";
}

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	
	int n, m, s;
	
	cin >> n >> m >> s;
	vector<Vertex> graph[SIZE];
	for (int i = 0; i<m; i++){
		Vertex a, b;
		cin >> a >> b;
		graph[a].push_back(b);
		graph[b].push_back(a);
	}
	
	cout << bida(graph, n);
}

File: tests/resources/bid/1/prog/bid.py
def bida(graf):
    for w in graf:
        if len(w) % 2 != 0:
            return "NIE"
    return "TAK"


def main():
    n, m, s = map(int, input().split())
    graf = [[] for _ in range(n + 1)]
    for _ in range(m):
        a, b = map(int, input().split())
        graf[a].append(b)
        graf[b].append(a)

    print(bida(graf))


if __name__ == '__main__':
    main()
